core/src/test/java/org/apache/iceberg/puffin/TestPuffinReader.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.puffin;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.iceberg.io.InMemoryInputFile;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Streams;
+import org.apache.iceberg.util.ByteBuffers;
+import org.apache.iceberg.util.Pair;
+import org.junit.Test;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.iceberg.puffin.PuffinCompressionCodec.NONE;
+import static org.apache.iceberg.puffin.PuffinCompressionCodec.ZSTD;
+import static org.apache.iceberg.puffin.PuffinFormatTestUtil.EMPTY_PUFFIN_UNCOMPRESSED_FOOTER_SIZE;
+import static org.apache.iceberg.puffin.PuffinFormatTestUtil.readTestResource;
+import static org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap.toImmutableMap;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestPuffinReader {
+  @Test
+  public void testEmptyFooterUncompressed() throws Exception {
+    testEmpty("v1/empty-puffin-uncompressed.bin", EMPTY_PUFFIN_UNCOMPRESSED_FOOTER_SIZE);
+  }
+
+  @Test
+  public void testEmptyWithUnknownFooterSize() throws Exception {
+    testEmpty("v1/empty-puffin-uncompressed.bin", null);
+  }
+
+  private void testEmpty(String resourceName, @Nullable Long footerSize) throws Exception {
+    InMemoryInputFile inputFile = new InMemoryInputFile(readTestResource(resourceName));
+    Puffin.ReadBuilder readBuilder = Puffin.read(inputFile)
+        .withFileSize(inputFile.getLength());
+    if (footerSize != null) {
+      readBuilder = readBuilder.withFooterSize(footerSize);
```

Comment:
Is it worth testing the cases where the footer size is wrong?

Reply:
good idea, added